app/js/app/directives/react_classes/GlossaryTermBlock.js

```diff
@@ -0,0 +1,79 @@
+define(["react", "jquery"], function(React,$) {
+    return function(ContentEditor, Block, ContentBlock) {
+        return React.createClass({
+
+            onDocChange: function(c, oldDoc, newDoc) {
+                this.props.onChange(this, oldDoc, newDoc);
+            },
+
+            onContentChange: function(e) {
+                var oldDoc = this.props.doc;
+                var newDoc = $.extend({}, oldDoc);
+                newDoc.value = e.target.value;
+                if (newDoc.autoId) {
+                    newDoc.id = e.target.value.toLowerCase().replace(/[^\w]/g, '-');
+                }
+                this.onDocChange(this, oldDoc, newDoc);
+            },
+
+            onExplanationChange: function(c, oldVal, newVal) {
+                var oldDoc = this.props.doc;
+                var newDoc = $.extend({}, oldDoc);
+                newDoc.explanation = newVal;
+                this.onDocChange(this, oldDoc, newDoc);
+            },
+
+            onAutoIdToggle: function(c, oldVal, newVal) {
+                var oldDoc = this.props.doc;
+                var newDoc = $.extend({}, oldDoc);
+                newDoc.autoId = !newDoc.autoId;
+                if (newDoc.autoId) {
+                    newDoc.id = newDoc.value.toLowerCase().replace(/[^\w]/g, '-');
+                }
+                this.onDocChange(this, oldDoc, newDoc);
+            },
+
+            onIdChange: function(c, oldVal, newVal) {
+                var oldDoc = this.props.doc;
+                var newDoc = $.extend({}, oldDoc);
+                newDoc.id = c.target.value.toLowerCase().replace(/[^\w]/g, '-');
+                this.onDocChange(this, oldDoc, newDoc);
+            },
+
+            toggle: function(attribute) {
+                return function() {
+                    var oldDoc = this.props.doc;
+                    var newDoc = $.extend({}, oldDoc);
+                    newDoc[attribute] = !oldDoc[attribute];
+                    this.onDocChange(this, oldDoc, newDoc);
+                }.bind(this);
+            },
+
+            render: function() {
+                if (typeof this.props.doc.autoId == 'undefined') {
+                    this.props.doc.autoId = true;
+                }
+                var emptyExplanation = {
+                    type: "content",
+                    children: [],
+                    encoding: "markdown"
+                };
+                return (
+                    <Block type="glossaryTerm" blockTypeTitle="Glossary term" doc={this.props.doc} onChange={this.onDocChange}>
+                        <div className="small-3 columns">
+                            <div className="row">
+                                <input type="text" value={this.props.doc.value} onChange={this.onContentChange} placeholder="Glossary term" />
```

Comment:
I think we should change the entry method to be more consistent with the other blocks, e.g. have a placeholder here, and change the explanation box on line 72 to be like the exposition boxes on the question pages - as we won't want it to be possible to put accordions (or other glossary terms) inside the glossary terms

Reply:
Done in 0e96f78624bfac65c3b441d04566275f5259580f.